components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -634,6 +635,10 @@ nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
         goto END;
     }
 
+    if (_busy && _parser.recv("SEND OK")) {
```

Comment:
Wait.. did the forum say that you receive "OK", not "SEND OK".

And my understanding was that `busy s` is coming before the `>`, this needs to be verified with AT debugs.

Reply:
Also, you cannot receive "busy" if you don't do `_process_oob(ESP8266_RECV_TIMEOUT, true);`

Reply:
Good catch, somehow I missed the recv() need. This will be much more difficult than it first seemed.

Reply:
Added more logic around this issue, please re-review